### PR TITLE
Scheme form

### PR DESCRIPTION
**Pull request by agentrickard**
Validation for the taxonomy scheme form. Includes work from #110.

### CODE REVIEW

File: src/Plugin/AccessControlHierarchy/Taxonomy.php [hunk 1]
```diff
@@ -353,9 +356,39 @@ public function buildConfigurationForm(array $form, FormStateInterface $form_sta
...
+      return [
+        'entity_type' => $entity_type,
+        'bundle' => $bundle,
+        'field' => $field_name,
+      ];
+    }, array_filter($settings['fields'])));
+    foreach ($settings['validate'] as $index => $value) {
+      $error = TRUE;
+      foreach ($settings['vocabularies'] as $vocabulary) {
+        if (in_array($vocabulary, $value)) {
```
Comment (larowlan) [inline src/Plugin/AccessControlHierarchy/Taxonomy.php]: nit:should pass the third argument here, they're strings

File: src/Plugin/AccessControlHierarchy/Taxonomy.php [hunk 2]
```diff
@@ -353,9 +356,39 @@ public function buildConfigurationForm(array $form, FormStateInterface $form_sta
...
+          $error = FALSE;
+        }
+      }
+      if ($error) {
+        $field = $form['fields']['#options'][$index];
+        $form_state->setErrorByName('scheme_settings][fields][' . $index, $this->t('The field @field on content type @type is not in the selected vocabularies.', ['@field' => $field['field'], '@type' => $field['bundle']]));
```
Comment (larowlan) [inline src/Plugin/AccessControlHierarchy/Taxonomy.php]: Should we add a test for this?

File: src/Plugin/AccessControlHierarchy/Taxonomy.php (hunk 1 above)
Comment (agentrickard) [reply src/Plugin/AccessControlHierarchy/Taxonomy.php]: I'm going to make that error forever.

File: src/Plugin/AccessControlHierarchy/Taxonomy.php (hunk 2 above)
Comment (agentrickard) [reply src/Plugin/AccessControlHierarchy/Taxonomy.php]: We should. It's fairly easy to do, on tap for tomorrow.

**Comment by agentrickard** [comment]
Updated, with a test.

**Pull request merged**